### ISSUE TITLE
Player Avatars and multiple Profiles

### ISSUE THREAD

**Issue by silelmot** [UltraStar-Deluxe/USDX#137]
Hello,
I don't know, if it is an issue or a suggestion... 

i tried to insert a profilepicture by editing of the config.ini and the [PLayerAvatar]-part
i tried it with P1=avatar.jpg, P1='avatar.jpg' and P1=./avatar.jpg... nothing happens.. what did i do wrong there?

and another thing... i often change playing ultrastar with my wife. till now we are changing the colour and name of player1 all the time ... is there a possibility to have more then 1 profile to just select?

**Comment by basisbit** [comment]
please make sure you are actually using the current UltraStar Deluxe version. You can download it at https://github.com/UltraStar-Deluxe/USDX/releases - when you select the sing menu, it will show you a players&avatar&difficulty selection page.

In case you have any questions left, feel free to ask here :smile:

**Comment by silelmot** [comment]
Sure. i have this. But there i have to type in my name on Player 1 one every time again, my girlfriend played and changed the name, and color.
There is no way to select Player "me" and player "girlfriend"... just if i play with 2 players i can select her and myself. but with only one player i always have to change everything again.


and there is no way to set an avatar. i am able to select the avatar-picture but it just changes to itself again. seems like there are no avatar-files.

**Comment by basisbit** [comment]
check that your installation is not in some folder with non-standard symbols like ß@ÖÄ§ or similar in the path name. I know there are some problems with that.

**Comment by basisbit** [comment]
also please provide screenshots

**Comment by silelmot** [comment]
i should say, that i am using this on linux.
the folder USDX exists within the GAME-Folder within the Avatars,
but in the game there is no avatar at all... 

its not that easy now with screenshots, because the game is just running on my steambox, and not on my main pc. i will try later. but there isnt anything to see at all....

**Comment by laird18** [comment]
I think silelmot raises two issues here, that I'm also having a problem with.

Firstly, a question about how to properly use the avatars.  I'm running USDX 1.3.2 on Ubuntu 16.04.  On the avatar selection page I don't see any avatars to chose from.  Is there a directory that I should put avatar images in?  I have tried to add absolute paths to the config.ini, as below, but no image appears.

[PlayerAvatar]
P1=/home/laird18/USDX/avatar.jpg

And secondly, a feature request.  It would be great if the user could have the ability to save a number of player profiles (i.e., an avatar, username, difficulty level) and the ability to select from a list of these at the player selection screen.  As I understand it at the moment, if users change player/microphone, you have to re-type in all the user names, re-select avatars etc... very inconvenient!

**Comment by basisbit** [comment]
the values behind the "PX=" in the config file are index values for cached versions of the images in the avatar.db file.
Add the avatar image file into the avatar folder (maybe @AlexanderS knows the default folder for linux based installs) and then select the image in player selection screen in the game.

**Comment by laird18** [comment]
Thanks for the reply @basisbit.

If @AlexanderS is able to help out that would be great.  This would be a great feature if I could get it working.

Congratulations on all the other great improvements in 1.3.2!

**Comment by AlexanderS** [comment]
I have no idea of the avatar stuff.

**Comment by AlexanderS** [comment]
@laird18 Looking at the [source code](https://github.com/UltraStar-Deluxe/USDX/blob/master/src/base/UPathUtils.pas#L182), it seems that the default avatars path is `/usr/share/ultrastardx/avatars/`.

**Comment by laird18** [comment]
Thanks @AlexanderS!  On inspection I discovered that the SharedPath on my installation was actually /usr/local/share/ultrastardx.  I created an "avatars" directory in there, put a few images in and now it's working!

A few observations.  Obviously in the linux version the default avatars aren't being installed for some reason (this should perhaps be reported as a bug?).  And I think it would be a good enhancement to make the avatar path customisable in config.ini just like the SongsDirs etc.

Thanks again for your help.

**Comment by AlexanderS** [comment]
Yes, avatars seems to be missing in install-data in the Makefile: https://github.com/UltraStar-Deluxe/USDX/blob/master/Makefile.in#L253

I will prepare a pull request later.
